janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/step/JanusGraphVertexStep.java

```diff
@@ -195,25 +193,53 @@ private void setParentMultiQueryStep() {
         }
 
         if (batchPropertyPrefetching) {
-            Set<Vertex> vertices = new HashSet<>();
-            result.forEach(v -> {
-                if (vertices.size() < txVertexCacheSize ) {
-                    vertices.add((Vertex) v);
-                }
-            });
```

Comment:
If I am understanding correctly, you are trying to optimize this for loop. Shall we do this instead?

```java
for (Vertex v : result) {
      if (vertices.size() < txVertexCacheSize ) {
            vertices.add((Vertex) v);
      } else {
            break;
      }
}
```

Reply:
Updated the PR